File: python/leetcode/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []

        def _traverse(root, res):
            if root:
                _traverse(root.left, res)
                res.append(root.val)
                _traverse(root.right, res)
            return res

        _traverse(root, result)
        return result

File: c#/leetcode/94-binary-tree-inorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        List<int> result = new List<int>();
        
        void traverse(TreeNode node) {
            if (node != null) {
                traverse(node.left);
                result.Add(node.val);
                traverse(node.right);
            }
        }
        
        traverse(root);
        return result;
    }
    
}